sqlite_fdw.c

```diff
@@ -5690,3 +5718,279 @@ sqliteIsForeignRelUpdatable(Relation rel)
+		}
+		else
+		{
+			/* error occurred in a scan against a foreign join */
+			TargetEntry *tle;
+
+			tle = list_nth_node(TargetEntry, fsplan->fdw_scan_tlist,
+								errpos->cur_attno - 1);
+
+			/*
+			 * Target list can have Vars and expressions.  For Vars, we can
+			 * get some information, however for expressions we can't.  Thus
+			 * for expressions, just show generic context message.
+			 */
+			if (IsA(tle->expr, Var))
+			{
+				Var		   *var = (Var *) tle->expr;
+
+				varno = var->varno;
+				colno = var->varattno;
+			}
+		}
+
+		if (varno > 0)
+		{
+			EState	   *estate = fsstate->ss.ps.state;
+			RangeTblEntry *rte = exec_rt_fetch(varno, estate);
+
+			relname = rte->eref->aliasname;
+
+			if (colno == 0)
+				is_wholerow = true;
+			else if (colno > 0 && colno <= list_length(rte->eref->colnames))
+				attname = strVal(list_nth(rte->eref->colnames, colno - 1));
+			else if (colno == SelfItemPointerAttributeNumber)
+				attname = "ctid";
+		}
+	}
+	else if (rel)
+	{
+		/* Non-ForeignScan case (we should always have a rel here) */
+		TupleDesc	tupdesc = RelationGetDescr(rel);
+
+		relname = RelationGetRelationName(rel);
+		if (errpos->cur_attno > 0 && errpos->cur_attno <= tupdesc->natts)
+		{
+			Form_pg_attribute attr = TupleDescAttr(tupdesc,
+												   errpos->cur_attno - 1);
+
+			attname = NameStr(attr->attname);
+		}
+		else if (errpos->cur_attno == SelfItemPointerAttributeNumber)
+			attname = "ctid";
+	}
+
+	{
+		/*
+		 * Error HINT block
+		 */
+		char	   *err_hint_mess0 = palloc(max_logged_byte_length * 2 + 1024); /* The longest hint message */
+		char 	   *err_hint_mess;
+		char	   *value_text = NULL;
+		bool		sqlite_value_as_hex_code = value_byte_size_blob_or_utf8 < max_logged_byte_length && ((GetDatabaseEncoding() != PG_UTF8 && value_aff == SQLITE3_TEXT) || (value_aff == SQLITE_BLOB));
+
+		/* Print problem SQLite values only for
+		 * - integer,
+		 * - float,
+		 * - short text if database encoding is UTF-8
+		 *   incorrect output otherwise possible: UTF-8 in SQLite, but not supported charcters in PostgreSQL
+		 */
+		if ((value_byte_size_blob_or_utf8 < max_logged_byte_length && GetDatabaseEncoding() == PG_UTF8 && value_aff == SQLITE3_TEXT)
+			|| value_aff == SQLITE_INTEGER
+			|| value_aff == SQLITE_FLOAT)
+			value_text = (char *)sqlite3_value_text(errpos->val);
+
+		if (sqlite_value_as_hex_code)
+		{
+			const unsigned char *vt = sqlite3_value_text(errpos->val);
+			value_text = palloc (max_logged_byte_length * 2 + 1);
+			for (size_t i = 0; i < value_byte_size_blob_or_utf8; ++i)
+				sprintf(value_text + i * 2, "%02x", vt[i]);
+	    }
+
+		err_hint_mess = err_hint_mess0;
+		if (value_text != NULL)
+		{
+			if (sqlite_value_as_hex_code)
+				err_hint_mess += sprintf(
+						err_hint_mess,
+						"SQLite hex value %s\n",
+						value_text );
+			else if (value_aff != SQLITE_INTEGER && value_aff != SQLITE_FLOAT)
+				err_hint_mess += sprintf(
+						err_hint_mess,
+						"SQLite value '%s'\n",
```

Comment:
Could you remove `\n`? I think it does not need to break line in hint/context message.

Reply:
SQLite value can be long (up to `NAMEDATALEN * 4`  in _hex_ case), hence additional part of hint (affinity and sometimes data length) will not very useful for reading. How do you think, @nxhai98 ?

Reply:
I think it no problem, `NAMEDATALEN * 4` in default config just have about 256 characters (3~4 terminal line) and it may an unusual case.   
At most cases in our test, this line to too short.

Reply:
Thanks for idea, @nxhai98 !  A good readable version without `\n`: `SQLite value with "" affinity` than `(%d bytes)` if applicable, than `in hex` if printable in hex and the original long value at the end. Changed to this message in https://github.com/pgspider/sqlite_fdw/pull/91/commits/311b818fdde67cd0bc5b45ac600d7359cfcc6d6f, please verify.

Reply:
Thank you. Please help me resolve this thread.